### DISCUSSION TITLE
Two richlet BCs with different values

### DISCUSSION THREAD

**Discussion by Ali1990dashti** [lululxvi/deepxde#1155] [General]
Dear community,
I just recently started using DeepXDE and appreciate your support.
I want to include two `Dirichlet BCs`. I followed [this discussion](https://github.com/lululxvi/deepxde/issues/104) but still I cannot solve my issue. I have a 1D rod and want to model the heat conduction. These are my `BCs`:
```
t_left = 100
t_right = 0
L = 1 # length of the rod
def boundary_r(x, on_boundary):
    return on_boundary and np.isclose(x[0], 0)
def boundary_l(x, on_boundary):
    return on_boundary and np.isclose(x[0], L)
def func_l(x):
    return t_left * np.ones((len(x),1),dtype = np.float32)
def func_r(x):
    return t_right * np.ones((len(x),1),dtype = np.float32)
bc_l = dde.icbc.DirichletBC(geom, func_l, boundary_l)
bc_r = dde.icbc.DirichletBC(geom, func_r, boundary_r)
ic = dde.icbc.IC(
    geomtime,
    lambda x: 50,
    lambda _, on_initial: on_initial,
)
# Define the PDE problem and configurations of the network:
data = dde.data.TimePDE(
    geomtime,
    pde,
    [bc_l, bc_r, ic],
    num_domain=2540,
    num_boundary=80,
    num_initial=160,
    num_test=2540,
)
```
But obiously the network is not working correctly. For example, when I run the model one of the losses is `nan`!:
```
Train loss
[2.93e-02, nan, 9.98e+03, 2.48e+03]
```
Thanks for your help and sorry for my trivial question.
Cheers
Ali

**Comment by lululxvi** [comment]
The BC code looks OK. The issue is due to something else.

**Comment by zenboprod** [comment]
Hello @Ali1990dashti, I also just recently started using DeepXDE. 
I am trying to include two different Neumann boundary conditions and I get the same problem as you.

Have you find the way to fix it ?

Thanks for your help 

Enzo

**Reply by zenboprod** [reply]
I found the issue.
In BC you should have geomtime, instead of geom.
